packages/zkdb/src/merkle-tree/merkle-tree-stream.ts

```diff
@@ -72,32 +76,34 @@ export class MerkleTreeWriteStream extends Writable {
     super({ objectMode: true });
   }
 
+  private isJSONParsed = false;
+
   _write(
     chunk: any,
     encoding: string,
     callback: (_?: Error | null) => void
   ): void {
     try {
-      if (typeof chunk === 'string') {
-        let parsed;
-        try {
-          parsed = JSON.parse(chunk);
-        } catch {
-          parsed = null;
-        }
+      if (Buffer.isBuffer(chunk)) {
+        this.buffer = Buffer.concat([this.buffer, chunk]);
```

Comment:
Check this implement, https://github.com/orochi-network/zkDatabase/blob/main/packages/zkdb/src/utilities/binary.ts#L72. `Uint8Array` is more universal to deal with binary data in JS.

Reply:
Fixed this